modules/apps/27-interchain-accounts/keeper/account.go

```diff
@@ -40,23 +40,28 @@ func (k Keeper) InitInterchainAccount(ctx sdk.Context, connectionID, counterpart
 	return nil
 }
 
-// Register interchain account if it has not already been created
-func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, portID string) {
-	address := types.GenerateAddress(portID)
+// RegisterInterchainAccount attempts to create a new account using the provided address and stores it in state keyed by the provided port identifier
+// If an account for the provided address already exists this function returns nil
+func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, address, portID string) error {
+	accAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return err
+	}
 
-	account := k.accountKeeper.GetAccount(ctx, address)
-	if account != nil {
-		return
+	if acc := k.accountKeeper.GetAccount(ctx, accAddr); acc != nil {
+		return nil
 	}
 
 	interchainAccount := types.NewInterchainAccount(
-		authtypes.NewBaseAccountWithAddress(address),
+		authtypes.NewBaseAccountWithAddress(accAddr),
 		portID,
 	)
 
 	k.accountKeeper.NewAccount(ctx, interchainAccount)
 	k.accountKeeper.SetAccount(ctx, interchainAccount)
```

Comment:
I think before we set the account in the state we need to do a check to ensure that the address passed into the version string by the relayer is the one that we expect (This is based on the counterparty (Controller) port-id). 

We want to ensure that a random relayer can not finish the handshake and choose a different address as the relayer address. @colin-axner

Reply:
I guess we can do this by calling `GenerateAddress(portID)` here and checking if the return value matches the address in the passed in Version string.

Reply:
Will add an additional check here against `types.GenerateAddress(portID)` and include a test